### DISCUSSION TITLE
Bosch BTH-RA (Radiator Thermostat II) => add install code to Zigbee 3.0 device

### DISCUSSION THREAD

**Discussion by hscheele** [Koenkk/zigbee2mqtt#16179] [Devices]
Hi all, 

Just trying to connect the Bosch Radiator Thermostat II, what works well with the original Smart Home Controller, but I want to get rid of it to access those Thermostats dirctly within Home Assistant without another Controller in the middle. 

As it is mentioned [here](https://www.zigbee2mqtt.io/devices/BTH-RA.html) that connenction should work, it does not work to me: 

- Zigbee2MQTT in version: 1.29.2-11.29.2-1 
- Home Assistant 2023.1.4 / Supervisor 2022.12.1 / Operating System 9.4 / Frontend 20230110.0 - latest (as virtual machine, running on my Synology DS220+)

Following are already working and were recognized properly: 

- Xiaomi - Aqara Smart Radiator Thermostat E1 (SRTS-A01)
- Xiaomi - Aqara temperature, humidity and pressure sensor (WSDCGQ11LM)
- Osram Smart+ plug (AB3257001NJ)
- IKEA TRADFRI signal repeater (E1746)

So my question: **What could be the next entry point to step into a clear analysis?** Please keep in mind, that I'm new to Zigbee2MQTT just trying to step over from Deconz so you cannot expect further detailed information at this point. Sorry to say! 

Thanks! 

// Hartmut

**Comment by hscheele** [comment]
Hi all, 

Since above, I found out that this might be something what every Zigbee 3.0 devices struggles with:  Add an install code to the coordinator! 

This was mentioned in [this issue report](https://github.com/Koenkk/zigbee2mqtt/issues/15595), where it was linked to this instruction: https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request-install-code-add

Problem to me is, how to do this within my HA Installation: 
- Does anyone have a short hint where to start with this? 
- Is there already an explanation available as I would expect that this is a rather common topic? 

// Hartmut

**Comment by g-jaeger** [comment]
Hi Hartmut,
Not sure if I got your issue. To make my Bosch device 'work', I had to scan the QR-code (was in the battery slot here) and paste the result to the 'add install code' tool (Settings -> Tools). Then it succeeded the interview phase.

Regards,
 Gerhard

**Reply by hscheele** [reply]
Hi Gerhard, 

I'm so thankful about your hint how to add the install code, because THIS WORKED! ;-)

Also thinking about what was the step that I missed to find this information as I would expect that most new users will face this issue at a time. May be it makes sense to add a comment in this instruction ([as stated before](https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request-install-code-add)). **But for my open case, I'm good for now!**
